sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvaluatePython.scala

```diff
@@ -120,10 +121,18 @@ object EvaluatePython {
     case (c: java.math.BigDecimal, dt: DecimalType) => Decimal(c, dt.precision, dt.scale)
 
     case (c: Int, DateType) => c
```

Comment:
BTW, as a side note, I think we can make the converter for the type and then reuse it.

Reply:
Of course, separate change obviously.